lib/fontist/errors.rb

```diff
@@ -9,10 +10,56 @@ class LicensingError < GeneralError; end
     class ManifestCouldNotBeFoundError < GeneralError; end
     class ManifestCouldNotBeReadError < GeneralError; end
     class MissingAttributeError < GeneralError; end
-    class MissingFontError < GeneralError; end
-    class NonSupportedFontError < GeneralError; end
     class TamperedFileError < GeneralError; end
     class TimeoutError < GeneralError; end
     class UnknownFontTypeError < GeneralError; end
+
+    class FontError < GeneralError
+      attr_reader :font, :style
+
+      def initialize(msg, font = nil, style = nil)
+        @font = font
+        @style = style
+
+        super(msg)
+      end
+
+      def name
+        messages = []
+        messages << "Font name: '#{@font}'"
+        messages << "Style: '#{@style}'" if @style
+        messages.join("; ")
+      end
+    end
+
+    class MissingFontError < FontError
+      def initialize(font, style = nil)
+        name = prepare_name(font, style)
+        msg = "#{name} font is missing, please run `fontist install '#{font}'` to download the font."
+
+        super(msg, font, style)
+      end
+
+      private
+
+      def prepare_name(font, style)
+        names = []
+        names << "'#{font}'"
+        names << "'#{style}'" if style
+        names.join(" ")
+      end
+    end
+
+    class NonSupportedFontError < FontError
```

Comment:
Can we rename this to be the `UnsupportedFontError`?

Reply:
Of course! Let's correct all these `non` and `un` prefixes.